### PR TITLE
Aerogear 9890

### PR DESCRIPTION
**Pull request by Boomatang**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-9890

## What
Add logging to the server unit of the MDC

## Why
To allow clear logging messages for the users

## How
By adding the winston js module to the package.json file

## Verification Steps
1. run `npm install` to get the latest packages
2. run the MDC both a production and non-production set up
3. view the logs and they show be a cleaner format
	`<timestamp(ISO 8601)>: <module> - <log level>: <Log message>`
4. in debug mode log file should be created locally
5. run `npm run lint`. There should be no warnings or errors

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task
- [ ] TODO

## Additional Notes

PS.: Add images and/or .gifs to illustrate what was changed if this pull request modifies the appearance/output of something presented to the users.

### CODE REVIEW

**Review by ziccardi — CHANGES_REQUESTED** [review]

File: server/logger.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        level: 'debug',
+        handleExceptions: true,
+        format: format.combine(format.timestamp(), format.label({ label: fileName }), messageFormat)
+      }
+    }
+  };
+
+  const logger = createLogger();
+
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (ziccardi) [inline server/logger.js]: We will log to the console most of the time, so I would configure the `file transport` as optional
```suggestion
 if (process.env.NODE_ENV === 'development') {
    logger.add(new transports.Console(options.debug.console));
    logger.add(new transports.File(options.debug.file)); // creates local dev-debug.log file
  } else {
    logger.add(new transports.Console(options.info.console));
  }
```

Why do we have a `options.debug.console` and a `options.info.console` ? They looks identical

**Review by ziccardi — CHANGES_REQUESTED** [review]

File: server/logger.js (hunk 1 above)
Comment (Boomatang) [reply server/logger.js]: Ok I will change this. I just want to point out that the `logger.verbose` calls will only show in the `debug` config. If required at a later stage we could add an extra flag for this.

Comment (Boomatang) [reply server/logger.js]: I missed your question about the `options.debug.console` and options.info.console`. The difference is the log level which is shown in the console. `options.debug.console` will log all levels from `debug` up while `options.info.console` only logs from `info` up which means `debug` and `verbose` are not logged to the console.

**Comment by Boomatang** [comment]
Closing

**Pull request closed without merging**